### PR TITLE
automation: Update help to indicate the relevance of job order

### PR DESCRIPTION
**Pull request by kingthorin**
## Overview
- CHANGELOG > Added change note.
- automation.html > Added new level three section within "Jobs" covering the importance of Job order.

## Related Issues
- Fixes zaproxy/zaproxy#8675

## Checklist
- [x] Update help
- [x] Update changelog
- [x] Run `./gradlew spotlessApply` for code formatting
- [na] Write tests
- [na] Check code coverage
- [x] Sign-off commits
- [x] Squash commits
- [x] Use a descriptive title

### CODE REVIEW

File: addOns/automation/src/main/javahelp/org/zaproxy/addon/automation/resources/help/contents/automation.html [hunk 1]
```diff
@@ -68,6 +68,14 @@ <H2>Jobs</H2>
 <li><a href="job-requestor.html">requestor</a> - crafts specific requests to send to the corresponding targets</li>
 <li><a href="job-ascan.html">activeScan</a> - runs the active scanner</li>
 </ul>
+
+<H3>Importance of Job Order</H3>
+The order of jobs is relevant and important. For example:
```
Comment (thc202) [inline addOns/automation/src/main/javahelp/org/zaproxy/addon/automation/resources/help/contents/automation.html:73]: Say also that the jobs are executed in the order they are defined in the plan? Maybe this should be said beforehand when explaining what a plan is and jobs are.

**Comment by kingthorin** [comment]
Tweaked

File: addOns/automation/src/main/javahelp/org/zaproxy/addon/automation/resources/help/contents/automation.html [hunk 2]
```diff
@@ -32,6 +32,8 @@ <H1>Automation Framework</H1>
 <li>Run the file using the <code>-autorun</code> commandline option e.g. <code>./zap.sh -cmd -autorun config.yaml</code>
 </ol>
 
+<strong>Note:</strong> The Jobs are executed in the order in which they appear (top to bottom) within the Plan .
```
Comment (ricekot) [inline addOns/automation/src/main/javahelp/org/zaproxy/addon/automation/resources/help/contents/automation.html:35]:
```suggestion
<strong>Note:</strong> The Jobs are executed in the order in which they appear (top to bottom) within the Plan.
```

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

Has a conflict.

**Review by psiinon — APPROVED** [review]

**Pull request merged**